src/controllers/commands/builtin/twitch/Prediction.cpp

```diff
@@ -66,4 +66,147 @@ QString createPrediction(const CommandContext &ctx)
     return "";
 }
 
+QString lockPrediction(const CommandContext &ctx)
+{
+    if (ctx.twitchChannel == nullptr)
+    {
+        const auto err = QStringLiteral(
+            "The /lockprediction command only works in Twitch channels");
+        if (ctx.channel != nullptr)
+        {
+            ctx.channel->addSystemMessage(err);
+        }
+        else
+        {
+            qCWarning(chatterinoCommands) << "Invalid command context:" << err;
+        }
+        return "";
+    }
+
+    auto currentUser = getApp()->getAccounts()->twitch.getCurrent();
+    if (currentUser->isAnon())
+    {
+        ctx.channel->addSystemMessage(
+            "You must be logged in to lock a prediction!");
+        return "";
+    }
+
+    const auto roomId = ctx.twitchChannel->roomId();
+    getHelix()->getPredictions(
+        roomId, {}, 1, {},
+        [channel = ctx.channel, roomId](const auto &result) {
+            if (result.predictions.empty())
+            {
+                channel->addSystemMessage("Failed to find any predictions");
+                return;
+            }
+
+            auto prediction = result.predictions.front();
+            if (prediction.status != "ACTIVE")
+            {
+                channel->addSystemMessage(
+                    "Could not find an active prediction");
+                return;
+            }
+
+            getHelix()->endPrediction(
+                roomId, prediction.id, false, {},
+                [channel](const HelixPrediction &data) {
+                    int points = 0;
+                    int users = 0;
+                    for (const auto &outcome : data.outcomes)
+                    {
+                        points += outcome.channelPoints;
+                        users += outcome.users;
+                    }
+
+                    channel->addSystemMessage(
+                        QString("Locked prediction with %1 points wagered by "
+                                "%2 users: '%3'")
+                            .arg(QString::number(points),
```

Comment:
Shall i obtain the current locale to format with commas

Reply:
You can if you'd like - no requirement